### PR TITLE
Http mock update

### PR DESCRIPTION
**Pull request by pragnagopa**
PR  https://github.com/Azure/azure-functions-java-library/pull/60 udpated HttpResponseMessage API.
This PR is updating the archetype to consume API changes

### CODE REVIEW

**Comment by pragnagopa** [comment]
Note: PR is not merged into library repo yet due to github issues.

**Comment by pragnagopa** [comment]
Library changes merged in and published to sonatype repo
https://oss.sonatype.org/content/repositories/commicrosoftazure-2669

File: azure-functions-archetype/src/main/resources/archetype-resources/pom.xml [hunk 1]
```diff
@@ -15,13 +15,28 @@
...
-        <azure.functions.java.library.version>1.0.0-beta-5</azure.functions.java.library.version>
+        <azure.functions.java.library.stagingrepo>commicrosoftazure-2669</azure.functions.java.library.stagingrepo>
+        <azure.functions.java.library.version>1.0.0-beta-6</azure.functions.java.library.version>
         <functionAppName>${appName}</functionAppName>
         <functionAppRegion>${appRegion}</functionAppRegion>
         <stagingDirectory>${project.build.directory}/azure-functions/${functionAppName}</stagingDirectory>
         <functionResourceGroup>${resourceGroup}</functionResourceGroup>
     </properties>
 
+    <repositories>
```
Comment (jdneo) [inline azure-functions-archetype/src/main/resources/archetype-resources/pom.xml]: Does the `<repositories>` attribute only exist for test purpose?

**Comment by pragnagopa** [comment]
Added Snapshot repository. Please publish snapshot version of the archetype and maven plugin with the latest changes and update the versions as well

**Comment by pragnagopa** [comment]
I have tested project builds with latest bits from archetype, plugin and azure-functions-java-library

**Comment by pragnagopa** [comment]
Addressed CR. Please review

File: azure-functions-archetype/src/main/resources/archetype-resources/pom.xml [hunk 2]
```diff
@@ -14,13 +14,27 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <maven.compiler.source>1.8</maven.compiler.source>
         <maven.compiler.target>1.8</maven.compiler.target>
-        <azure.functions.maven.plugin.version>1.0.0-beta-7</azure.functions.maven.plugin.version>
-        <azure.functions.java.library.version>1.0.0-beta-5</azure.functions.java.library.version>
+        <azure.functions.maven.plugin.version>1.0.0-beta-7</azure.functions.maven.plugin.version>
```
Comment (jdneo) [inline azure-functions-archetype/src/main/resources/archetype-resources/pom.xml]: tailing spaces are not needed.

File: azure-functions-archetype/src/main/resources/archetype-resources/pom.xml [hunk 3]
```diff
@@ -14,13 +14,27 @@
...
-        <azure.functions.java.library.version>1.0.0-beta-5</azure.functions.java.library.version>
+        <azure.functions.maven.plugin.version>1.0.0-beta-7</azure.functions.maven.plugin.version>       
+        <azure.functions.java.library.version>1.0.0-beta-7-SNAPSHOT</azure.functions.java.library.version>
         <functionAppName>${appName}</functionAppName>
         <functionAppRegion>${appRegion}</functionAppRegion>
         <stagingDirectory>${project.build.directory}/azure-functions/${functionAppName}</stagingDirectory>
         <functionResourceGroup>${resourceGroup}</functionResourceGroup>
     </properties>
+	
+	<repositories>
```
Comment (jdneo) [inline azure-functions-archetype/src/main/resources/archetype-resources/pom.xml]: tab -> spaces here

File: azure-functions-archetype/src/main/resources/archetype-resources/pom.xml [hunk 4]
```diff
@@ -14,13 +14,27 @@
...
+			<name>Maven Central Snapshot Repository</name>
+			<url>https://oss.sonatype.org/content/repositories/snapshots/</url>
+			<releases>
+				<enabled>false</enabled>
+			</releases>
+			<snapshots>
+				<enabled>true</enabled>
+			</snapshots>
+		</repository>		
+	</repositories>
```
Comment (jdneo) [inline azure-functions-archetype/src/main/resources/archetype-resources/pom.xml]: A question not a review comment here:
Should we leave the <repositories> field as it is when we releasing to Maven Central? Or we need to remove it?

File: azure-functions-archetype/src/main/resources/archetype-resources/pom.xml (hunk 2 above)
Comment (pragnagopa) [reply azure-functions-archetype/src/main/resources/archetype-resources/pom.xml]: Which IDE do you recommend? Is there a settings file you can share?

File: azure-functions-archetype/src/main/resources/archetype-resources/pom.xml (hunk 4 above)
Comment (pragnagopa) [reply azure-functions-archetype/src/main/resources/archetype-resources/pom.xml]: Yes. You can leave it in here. For future releases SNAPSHOT versions can be referenced from here.

File: azure-functions-archetype/src/main/resources/archetype-resources/pom.xml (hunk 2 above)
Comment (jdneo) [reply azure-functions-archetype/src/main/resources/archetype-resources/pom.xml]: I'm using VS Code 😃 

You can install an extension called: `XML Tools`. Set the indention to spaces with size 4. And then trigger format command.

File: azure-functions-archetype/src/main/resources/archetype-resources/pom.xml (hunk 4 above)
Comment (pragnagopa) [reply azure-functions-archetype/src/main/resources/archetype-resources/pom.xml]: Thanks. Used VSCode!

**Pull request merged**